src/data/set/basic.lean

```diff
@@ -1290,6 +1296,10 @@ theorem forall_range_iff {p : α → Prop} : (∀ a ∈ range f, p a) ↔ (∀ i
 theorem exists_range_iff {p : α → Prop} : (∃ a ∈ range f, p a) ↔ (∃ i, p (f i)) :=
 ⟨assume ⟨a, ⟨i, eq⟩, h⟩, ⟨i, eq.symm ▸ h⟩, assume ⟨i, h⟩, ⟨f i, mem_range_self _, h⟩⟩
 
+lemma exists_range_iff' {p : α → Prop} :
+  (∃ a, a ∈ range f ∧ p a) ↔ ∃ i, p (f i) :=
+⟨by rintros ⟨b, ⟨a, rfl⟩, h⟩ ; exact ⟨a, h⟩, λ ⟨a, h⟩, ⟨f a, mem_range_self a, h⟩⟩
```

Comment:
Does `by simpa only [exists_prop] using exists_range_iff` work here? This way the difference between two lemmas is more visible.

Reply:
It does work. I don't really understand why you prefer this proof but I'm happy to change it (I don't really understand mathlib conventions, I'm only just beginning to make PR's on a regular basis). It's about 20ms slower ;-)

Reply:
Maybe I'm wrong here. My only reason was to make it clear that this is already done in `exists_range_iff`.